### ISSUE TITLE
Venue not showing up

### ISSUE THREAD

**Issue by jidulberger** [jmatheson/OnMe-Backend#73]
Highland Tap and Burger (near me) is not showing up in the app.

It's here on foursquare: foursquare.com/highlandtandb

I suspect it's due to the category we are using - the first category is "burger joint", followed by "sports bar" and "bar" .

![image](https://cloud.githubusercontent.com/assets/2204043/24877505/ca69cfae-1dec-11e7-82a7-7af4d4608756.png)

**Comment by volkzayaz** [comment]
I'd pass this to @jmatheson. He had some problems making one of his favorite's venues to show up. Eventually he did by specifying City. He wanted to play with Foursquare API to find optimal combination. 

Let's return to this ticket as soon as @jmatheson, comes up with solution for API

**Comment by jmatheson** [comment]
API still needs work

**Comment by jmatheson** [comment]
@volkzayaz I think this might be the same problem I am running in to.
In the old app we had a table of many different types of venues and then we were searching for venues that matched any of the tags. Maybe this did not make it across?

I noticed in the newest version that the coffee shops are a better listing that before but the overall list, beer etc was not.

Here is the Wayward Brewery that I was having the same problem with. Brewery was a type of beer venue in the old version of the app (it was stored in Parse on the server side) @volkzayaz 

https://foursquare.com/v/wayward-brewing-co/5590c329498e61fcb7cbfa8a

**Comment by volkzayaz** [comment]
dependent on #86

**Comment by volkzayaz** [comment]
@jmatheson, we double checked and you were right. We didn't reference all of the categories that were set up in the former `VenueCategories.json` table. This should be the issue for missing Venues.

Changes were related to backend only. Please test venue searching and let me know if it became better/worse/didn't change.

**Comment by jmatheson** [comment]
Can you let me know how you implemented this - was it done on the back end so we can change it in the future without app updates?

**Comment by jmatheson** [comment]
It looks worse. Can you please share the categories that are used for each tab?

The old version had ways to filter out fake venues. I am worried we are not doing that here, but it's hard to tell. I see venues with no address and some that show ratings of 0/10. No address used to be filtered for fake venues.

It looks like there is an issue with sort order too. When I go to coffee the list it chose to return is sorted by distance from me (seems wrong). It also no longer shows any of the coffee shops near me that are decent.

**Comment by jmatheson** [comment]
When i left wifi for 4g the venues in the main tab started to look better. It now shows the wayward brewery and the coffee shop by my house.

But the list under beer is totally different and does not include the wayward and the coffee tab does not show the coffee shop by my house. 

Is there different logic for the beer, wine, etc tabs?

**Comment by volkzayaz** [comment]
> Can you let me know how you implemented this - was it done on the back end so we can change it in the future without app updates?

Yes. VenueManager.php file is responsible for Venue requests.

> The old version had ways to filter out fake venues. I am worried we are not doing that here, but it's hard to tell. I see venues with no address and some that show ratings of 0/10. No address used to be filtered for fake venues.

Currently we do not do this. Konstantin will update logic to filter out malformed Venues.

> Is there different logic for the beer, wine, etc tabs?

No, code is the same, but differs in venue categories. Here's the snippet from Constants.php.

```
const VENUES_CATEGORY_BEER = '52e81612bcbc57f1066b7a06,4bf58dd8d48988d155941735,50327c8591d4c4b30a586d5d,4bf58dd8d48988d11d941735,4bf58dd8d48988d117941735,4bf58dd8d48988d116941735,4bf58dd8d48988d118941735,4bf58dd8d48988d11b941735';
 +    const VENUES_CATEGORY_COCKTAILS = '4bf58dd8d48988d121941735,4bf58dd8d48988d110941735,4bf58dd8d48988d1d4941735,52e81612bcbc57f1066b7a0d,4e0e22f5a56208c4ea9a85a0,4bf58dd8d48988d11e941735,4bf58dd8d48988d122941735';
 +    const VENUES_CATEGORY_WINE = '4bf58dd8d48988d110941735,4bf58dd8d48988d123941735,4bf58dd8d48988d14b941735,4bf58dd8d48988d119951735,4bf58dd8d48988d1cc941735';
 +    const VENUES_CATEGORY_COFFEE = '4bf58dd8d48988d1e0931735,4bf58dd8d48988d1dc931735,4bf58dd8d48988d16a941735,4bf58dd8d48988d147941735,4bf58dd8d48988d1e0931735';
 +    const VENUES_CATEGORY_ALL = '4bf58dd8d48988d1e0931735,4bf58dd8d48988d121941735,4bf58dd8d48988d110941735,4bf58dd8d48988d1d4941735,4bf58dd8d48988d1dc931735,52e81612bcbc57f1066b7a06,4bf58dd8d48988d123941735,4bf58dd8d48988d155941735,4bf58dd8d48988d14b941735,4bf58dd8d48988d16a941735,50327c8591d4c4b30a586d5d,4bf58dd8d48988d11d941735,4bf58dd8d48988d117941735,52e81612bcbc57f1066b7a0d,4e0e22f5a56208c4ea9a85a0,4bf58dd8d48988d11e941735,4bf58dd8d48988d116941735,4bf58dd8d48988d147941735,4bf58dd8d48988d118941735,4bf58dd8d48988d112941735,4bf58dd8d48988d11b941735,4bf58dd8d48988d119951735,4bf58dd8d48988d122941735,4bf58dd8d48988d1e0931735,4bf58dd8d48988d1cc941735,4bf58dd8d48988d11c941735';
```

**Comment by jmatheson** [comment]
OK, it looks like the right clusters of categories per your snippet above.

For some reason the lists are not consistent. Sometimes I open the app and nearby popular venues are there and sometimes they are not. Despite sitting in the same location. 

The only thing I can think it might be is
1) there is some sort of cache that is deciding to ringfence me at a wider distance
2) it's not pulling the right set of venues somehow thing distance needs to be wider than it really needs to be. I am not sure why that would be inconsistent though.

There only needs to be 3 or more venues across the entire array of constants for that tab in order to use the tighter radius.

**Comment by jidulberger** [comment]
The missing venue (highland tap and burger) I reported now shows up; I am not seeing anything else that's missing, but will continue to test.

**Comment by volkzayaz** [comment]
0.0.24 has improvements for location determining. Previous had issues, sometimes old location was treated as current which resulted in poor venues selection.
@jmatheson, @jidulberger, let me know if experience became better/worse/didn't change

**Comment by jmatheson** [comment]
Looks better. Will continue to test. I am surprised that it is now simply filtering from the full list. This may or may not work. For example if there are 25 venues from the "all" list but none of them are coffee, then what happens? @jidulberger can you test this too?

**Comment by volkzayaz** [comment]
> it is now simply filtering from the full list

It's not. It's a coincidence and an animation that makes such impression

**Comment by jidulberger** [comment]
Should we change the animation? Is that confusing?

Otherwise, I'm not seeing any counterintuitive results, it looks good to
me.

**Comment by jmatheson** [comment]
I am ok with the animation as is.

I would however like to ensure that if you tap on a different tab (beer vs wine for example) that it scrolls to the top of the list. Example, I open the app. Scroll down the page. Press Beer. Now I am scrolled down the page and it is not intuitive to me that I am now in the middle of the list instead of the top.

**Comment by jidulberger** [comment]
I hadn't noticed that, but looking at it now, I entirely agree: switching
tabs should scroll user to the top of the list.

**Comment by volkzayaz** [comment]
fixed in 0.0.26

**Comment by jmatheson** [comment]
fixed
